scip-java/src/main/scala/com/sourcegraph/scip_java/buildtools/MillBuildTool.scala

```diff
@@ -0,0 +1,110 @@
+package com.sourcegraph.scip_java.buildtools
+
+import java.nio.file.Files
+import java.nio.file.StandardCopyOption
+
+import scala.jdk.CollectionConverters._
+
+import com.sourcegraph.scip_java.commands.IndexCommand
+import com.sourcegraph.scip_java.BuildInfo
+
+class MillBuildTool(index: IndexCommand) extends BuildTool("mill", index) {
+
+  override def usedInCurrentDirectory(): Boolean =
+    Files.isRegularFile(index.workingDirectory.resolve("build.sc"))
+
+  override def generateScip(): Int =
+    millVersion() match {
+      case Some(version) if isSupportedMillVersion(version) =>
+        unconditionallyGenerateScip()
+      case Some(version) =>
+        failFast(
+          s"Unsupported Mill version '${version}'. " +
+            s"To fix this problem, upgrade Mill to at least ${minimalMillVersion} and try again."
+        )
+      case None =>
+        failFast(
+          s"No Mill version detected. " +
+            s"To fix this problem, run the following command and try again: " +
+            s"echo '${minimalMillVersion}' >> .mill-version"
+        )
+    }
+
+  private def failFast(message: String): Int = {
+    index.app.error(message)
+    1
+  }
+
+  private def unconditionallyGenerateScip(): Int = {
+    val localMill = Files.isRegularFile(millFile)
+    val command =
+      if (localMill) {
+        "./mill"
```

Comment:
This is also one thing I need to test first locally to ensure proc'ing like this with ./mill actually works. I figured it's a good idea to try and just use the local mill if they have one.

Reply:
Was able to verify this worked:

```
❯ cs launch com.sourcegraph:scip-java_2.13:0.8.2-1-53a602d2-SNAPSHOT -- index
$ ./mill --import ivy:io.chris-kipp::mill-scip::0.2.1 io.kipp.mill.scip.Scip/generate
Compiling /Users/ckipp/Documents/scala-workspace/mill-scip/build.sc
[1/1] io.kipp.mill.scip.Scip.generate
Found 1 total modules to process.
[1/1] io.kipp.mill.scip.Scip.generate > [12/44] plugin.generatedSources.overridden.mill.scalalib.JavaMo
[1/1] io.kipp.mill.scip.Scip.generate > [20/44] plugin.compileIvyDeps.overridden.mill.scalalib.JavaModu
[1/1] io.kipp.mill.scip.Scip.generate > [23/44] plugin.ivyDeps.overridden.mill.scalalib.JavaModule.ivyD
[1/1] io.kipp.mill.scip.Scip.generate > [25/44] plugin.mandatoryIvyDeps.overridden.mill.scalalib.JavaMo
[1/1] io.kipp.mill.scip.Scip.generate > [33/44] plugin.scalacPluginIvyDeps.overridden.mill.scalalib.Sca
[1/1] io.kipp.mill.scip.Scip.generate > [36/44] plugin.scalacOptions.overridden.mill.scalalib.ScalaModu
[1/1] io.kipp.mill.scip.Scip.generate > [37/44] plugin.scalacOptions.overridden.com.goyeau.mill.scalafi
[1/1] io.kipp.mill.scip.Scip.generate > [43/44] mill.scalalib.ZincWorkerModule.worker
Ensuring everything needed for semanticdb version [4.5.11] is available
Generating semanticdb for [mill-scip_mill0.10_2.13]
[info] compiling 5 Scala sources to /Users/ckipp/Documents/scala-workspace/mill-scip/out/io/kipp/mill/scip/Scip/generate.dest/classes ...
[info] Ignoring duplicate plugin semanticdb (scala.meta.internal.semanticdb.scalac.SemanticdbPlugin)
[info] done compiling
Creating a index.scip in /Users/ckipp/Documents/scala-workspace/mill-scip/out/io/kipp/mill/scip/Scip/generate.dest/index.scip
Including 130 classpath entries
Staring to create a scip index from 5 SemanticDB documents.
[1/1] io.kipp.mill.scip.Scip.generate | [5/5] Finished creating your scip index
info: /Users/ckipp/Documents/scala-workspace/mill-scip/index.scip
```